File: Day88/DSA/Find_the_XOR_of_Numbers_Which_Appear_Twice.cpp
Question:- https://leetcode.com/problems/find-the-xor-of-numbers-which-appear-twice/description/

Code:- 

class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (auto i : nums) {
            mp[i]++;
        }

        int ans = 0;
        for (auto i : mp) {
            if (i.second == 2) {
                ans = ans ^ i.first;
            }
        }

        return ans;
    }
};

File: Day88/DSA/Find_the_XOR_of_Numbers_Which_Appear_Twice.java
Question:- https://leetcode.com/problems/find-the-xor-of-numbers-which-appear-twice/

Code:- 

class Solution {
    public int duplicateNumbersXOR(int[] nums) {

        HashMap<Integer,Integer> mp = new HashMap<>();

        for (int i : nums) {
            int count = mp.getOrDefault(i,0);
            mp.put(i , count+1);
        }

        int ans = 0;

        for(Map.Entry<Integer, Integer> ele : mp.entrySet() )
        {
            int key = ele.getKey();
            int val = ele.getValue();

            if(val == 2)
            {
                ans ^= key;
            }
        }

        return ans;
    }
}